spec/abstract-jobs.html

```diff
@@ -222,11 +222,11 @@ <h1>Execution</h1>
           of non-live objects.
         </p>
         <p>
-          Implementations are obligated to empty WeakRefs for all objects in a
-          non-live set simultaneously. In other words, they must not empty a
-          WeakRef pointing to an object _obj_ without emptying out other
-          WeakRefs that, if not emptied, could result in an execution that
-          observes the Object value of _obj_.
+          If an implementation chooses a non-live set _S_ in which to empty WeakRefs, it
+          must empty WeakRefs for all objects in _S_ simultaneously. In other
+          words, they must not empty a WeakRef pointing to an object _obj_
```

Comment:
```suggestion
          words, an implementation must not empty a WeakRef pointing to an object _obj_
```
(the antecedent is now singular, so "they" is weird)

Reply:
singular they is perfectly correct; adding the word “implementation” here triggers the discussion about host vs implementation.

Reply:
Implementations are not animate enough to warrant "they". "it", at least.

Reply:
Sure, that works fine.

Reply:
I like @bakkot's suggestion. In my preferred taxonomy of host-implementation, implementation is the correct choice. Plus, I already start the paragraph with "If an implementation".

We can debate host-vs-implementation independently. No reason do it in this proposal spec draft.